src/IPC/QueueRx.cc

```diff
@@ -1,22 +1,74 @@
 #include <iostream>
 #include "IPC.h"
 
+#include "Queue.h"
+
+#include <sys/stat.h>   /* S_IRWXU S_IRWXG */
+#include <sys/ipc.h>    /* IPC_CREAT */
+#include <sys/msg.h>    /* msgget() */
+#include <string.h>     /* strerror() */
+
+char writeBuf[BUFFERSIZE];
+queuedMessage msg;
+
 void QueueRx::receive(void) {
     this->setupQueueRx();
 
-    this->fileSizeRx();
-
     this->queueRx();
 }
 
-void QueueRx::setupQueueRx(void) const {
-    ;
+void QueueRx::setupQueueRx(void) {
+    key_t key;
+
+    /* setup message queue */
+    key = ftok(PathName, ProjectId);    /* key to identify the queue */
+    if (key<0) {
+        throw std::runtime_error(strerror(errno));
+    }
+
+    this->qid = msgget(key, S_IRWXU | S_IRWXG | IPC_CREAT);
```

Comment:
Maybe would be nice to check if a queue already exist to close it and open it again empty, immagine the following scenario.

- I launch the sender
- I launch the receiver
- the copy didn't finished but the sender and the receiver both crashed without calling their destructors (maybe ctrl+c hapened)
- I relaunch the receiver that connects to the old queue because it wasn't destroyed.
Will it get false data from it ?